### PR TITLE
[Categories Redesign] - Display categories at the top

### PR DESCRIPTION
**Pull request by mebarbosa**
## Description
- This PR moves the categories from bottom to the top of Discover.
- Hides the search bar from carousel and displays above of categories.

This is part of the categories redesign. You can find references and figma here in our internal reference: `pdeCcb-4tv-p2`.

> [!note]
> This is not the final design. The new design will be implemented in different pull requests to make it easier to review

> [!note]
> This feature is behind the local feature flag `CATEGORIES_REDESIGN`

#### 1. With Feature flag OFF
1. You should see the Discover tab with the current implementation in prod without any change

#### 2. With Feature flag ON
1. The categories should be displayed at the top below of the search bar

## Screenshots or Screencast 
[Screen_recording_20240308_143124.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/3f03f52a-7fc8-435a-ad87-ae868d514ee3)

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverFragment.kt [hunk 1]
```diff
@@ -214,6 +220,20 @@ class DiscoverFragment : BaseFragment(), DiscoverAdapter.Listener, RegionSelectF
...
+            val categoriesIndex = mutableContentList.indexOfFirst { it is DiscoverRow && it.type is ListType.Categories }
+
+            if (categoriesIndex != -1) {
+                val categoriesItem = mutableContentList.removeAt(categoriesIndex)
+                mutableContentList.add(0, categoriesItem)
+            }
+        }
+
+        return mutableContentList
+    }
```
Comment (mebarbosa) [inline modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverFragment.kt:236]: I will sync with iOS and backend if there is a way to remotely set categories at the top without needing this so I can address it to another PR

File: modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverAdapter.kt [hunk 2]
```diff
@@ -489,6 +501,20 @@ internal class DiscoverAdapter(
...
+                        service.getCategoriesList(row.source),
+                        onSuccess = { categories ->
+                            val sortedCategories = categories.map { it.copy(name = it.name.tryToLocalise(resources)) }.sortedBy { it.name }
+
+                            adapter.submitList(sortedCategories) {
+                                onRestoreInstanceState(holder)
+                            }
+                        },
+                    )
+                }
```
Comment (mebarbosa) [inline modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverAdapter.kt:517]: I just copied the `CategoriesViewHolder`. The new design for categories will be addressed in another PR

**Comment by CookieyedCodes** [comment]
Is the plan for this to shrink categories? Unless it's expanded? Be careful guys with the amount of ads guys it can devalue the apps precived quality 😅

**Comment by mebarbosa** [comment]
> Is the plan for this to shrink categories? Unless it's expanded? Be careful guys with the amount of ads guys it can devalue the apps precived quality 😅

@CookieyedCodes The categories will not look like this 🤣. I agree that it's not good, but this is just a pull request to prepare for the redesign. I personally love the new design that will be implemented soon.

**Comment by CookieyedCodes** [comment]
Coolio,
Looking forward to seeing it ,if its it's like dynamic pills or something like that I'd probably be fine with it 🤔🤔

**Review by MiSikora — APPROVED** [review]

**Review by mukul-scogo — APPROVED** [review]
